src/python_testing/MinimalRepresentation.py

```diff
@@ -0,0 +1,141 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from dataclasses import dataclass, field
+
+from chip.tlv import uint
+from conformance_support import ConformanceDecision
+from global_attribute_ids import GlobalAttributeIds
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from TC_DeviceConformance import DeviceConformanceTests
+
+
+@dataclass
+class ClusterMinimalElements:
+    # masks
+    features: list[uint] = field(default_factory=list)
+    # IDs
+    attributes: list[uint] = field(default_factory=list)
+    # Only received commands are necessary - generated events are ALWAYS determined from accepted
+    commands: list[uint] = field(default_factory=list)
+    # TODO: need event support
+
+
+class MinimalRepresentationChecker(DeviceConformanceTests):
+    def GenerateMinimals(self, ignore_in_progress: bool, is_ci: bool) -> dict[uint, dict[uint, ClusterMinimalElements]]:
+        if not self.xml_clusters:
+            self.setup_class_helper()
+
+        success, _ = self.check_conformance(ignore_in_progress, is_ci)
+        if not success:
+            self.fail_current_test("Problems with conformance")
+
+        # Now what we know the conformance is OK, we want to expose all the data model elements on the device
```

Comment:
should you have a return then to "know that conformance is OK"? otherwise this runs even if conformance check failed.